src/async_impl/client.rs

```diff
@@ -76,8 +76,10 @@ pub struct Client {
 
 /// A `ClientBuilder` can be used to create a `Client` with custom configuration.
 #[must_use]
-pub struct ClientBuilder {
+pub struct ClientBuilder<CL1, CL2> {
     config: Config,
+    // separated out to simplify casting between generic types while copying config
+    connector_layers: ConnectorLayerBuilder<CL1, CL2>,
```

Comment:
Why does `ConnectorLayerBuilder` have 2 layers?

Reply:
My mental model was flawed here I think, we are holding two layers in the stack at a given time, but we can simplify the generic to use a single one with related bounds. Taking a stab at this.

Reply:
Pushed a change to simplify to a single generic bound, thanks for the callout.